Check for compatible node,npm versions on install

zpao opened the pull request:
This is only for building React and doesnn't apply to production installs.

We're going to switch to iojs shortly and I wanted to make sure we short-circuit people trying to develop React so we don't get reports of weird issues.

zpao commented:
cc @yungsters @cpojer. This is what I was thinking for Relay and fbjs. I'll bring this (or whatever we end up with) over to those projects, maybe sharable in fbjs.

cpojer commented:
that is nice! How do you feel about putting the info into package.json and use that instead of putting the version numbers into the script?

zpao commented:
Moved it into package.json as `devEngines` (I like to think this is a good parallel to `dependencies` ⇔ `devDependencies`). I also moved it to the `postinstall` hook so we can actually require semver. And then I also used some builtin node modules instead of writing my own assertions.

Here's what it looks like in action:
<img width="902" alt="screen shot 2015-08-18 at 9 57 27 pm" src="https://cloud.githubusercontent.com/assets/8445/9349122/fb0e6730-45f5-11e5-82b8-e75be20e3616.png">

cpojer commented:
`Current node version is not support for development`, what?

I love how you implemented `devEngines` half a day after I said this is what I would like to have. You are the best.

zpao commented:
Going to merge this and then we can figure out a longer term place for it (perhaps in fbjs) later.
